mysensors/mysensors.py

```diff
@@ -49,6 +50,22 @@ def __init__(self, event_callback=None, persistence=False,
         if persistence:
             self._safe_load_sensors()
 
+    @property
+    def event_callback(self):
+        """Getter for event_callback property."""
+        return self._event_callback
+
+    @event_callback.setter
+    def event_callback(self, callback):
+        """Setter for event_callback property."""
+        self._event_callback = callback
+        if callback is not None:
+            if len(inspect.signature(callback).parameters) == 2:
+                def adapter(message):
+                    """Event callback adapter for backwards compatibility."""
+                    callback('sensor_update', message.node_id)
+                self._event_callback = adapter
```

Comment:
Just to clarify what I mean: Yes, existing callback signatures are supported, but not calling the function stored in the `event_callback` attribute, with the old signature. This attribute is not hidden from the user, and I don't think it should be hidden either, so we should expect it to be called by the user.

I prefer a breaking change in this case.